lib/private/shell.ts

```diff
@@ -1,23 +1,23 @@
 import * as child_process from 'child_process';
+import { EventType, globalOutputHandler } from '../progress';
 
 export type Logger = (x: string) => void;
 
 export interface ShellOptions extends child_process.SpawnOptions {
   readonly quiet?: boolean;
-  readonly logger?: Logger;
   readonly input?: string;
 }
 
 /**
  * OS helpers
  *
- * Shell function which both prints to stdout and collects the output into a
- * string.
+ * Shell function which both emits the output to the configured output handler, and collects the output
+ * to return it as a string.
  */
 export async function shell(command: string[], options: ShellOptions = {}): Promise<string> {
-  if (options.logger) {
-    options.logger(renderCommandLine(command));
-  }
+  globalOutputHandler.publishEvent(EventType.START, command.join(' '));
+  globalOutputHandler.info(renderCommandLine(command));
```

Comment:
So from what I could gather, it used to be that this message would eventually be logged as:

https://github.com/cdklabs/cdk-assets/blob/df4fa2353ca190fe0c11f13e386553157ccd15de/lib/private/handlers/container-images.ts#L41

But now it changed to:

https://github.com/cdklabs/cdk-assets/blob/cacbcf4a7fb8af3107e43331c5f9c3bb95f497b9/lib/progress.ts#L125-L127

Is this what you meant to happen? Also, it used to be that this message was logged via the implenentation of `IHostHandler`:

https://github.com/cdklabs/cdk-assets/blob/cacbcf4a7fb8af3107e43331c5f9c3bb95f497b9/lib/publishing.ts#L116-L118

Which does some additional stuff apart from just logging. But now it just invokes the progress listener and it doesn't seem to be equivalent.